plugin-api/src/main/java/org/sonar/api/issue/IssueStatus.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Sonar Plugin API
+ * Copyright (C) 2009-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.api.issue;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @since 10.4
+ */
+public enum IssueStatus {
+
+
+  OPEN,
+
+  @Deprecated(since = "10.4")
+  /*
+   * @deprecated use {@link IssueStatus#ACCEPTED} instead
+   */
+    CONFIRMED,
+  FALSE_POSITIVE,
+  ACCEPTED,
+  FIXED;
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(IssueStatus.class);
+
+  @CheckForNull
+  public static IssueStatus of(@Nullable String status, @Nullable String resolution) {
+
+    //null status is not supposed to happen, but since it is nullable in database, we want the mapping to be resilient.
```

Comment:
this comment about the database doesn't make sense in the context of plugin-api, we may want to adapt it